### DISCUSSION TITLE
Question about CentralDifference Solver Implementation

### DISCUSSION THREAD

**Discussion by chunhuizhao478** [idaholab/moose#24243] [Q&A General] [answered]
Hi all, I'm a bit confused about the implementation in central difference solver and hope someone could clarify it. In the CentralDifference.h, we have the following code snippet for evaluating  ```u_dot``` and  ```u_dotdot```:
```
template <typename T, typename T2, typename T3, typename T4, typename T5>
void
CentralDifference::computeTimeDerivativeHelper(T & u_dot,
                                               T2 & u_dotdot,
                                               const T3 & u_old,
                                               const T4 & u_old_old,
                                               const T5 & u_old_old_old) const
{
  // computing first derivative
  // using the Central Difference method
  // u_dot_old = (first_term - second_term) / 2 / dt
  //       first_term = u
  //      second_term = u_older
  u_dot = u_old;
  u_dot -= u_old_old_old; // 'older than older' solution
  u_dot *= 1.0 / (2.0 * _dt);

  // computing second derivative
  // using the Central Difference method
  // u_dotdot_old = (first_term - second_term + third_term) / dt / dt
  //       first_term = u
  //      second_term = 2 * u_old
  //       third_term = u_older
  u_dotdot = u_old;
  u_dotdot -= u_old_old;
  u_dotdot -= u_old_old;
  u_dotdot += u_old_old_old;
  u_dotdot *= 1.0 / (_dt * _dt);
}
```
However, the ```u_old``` is extracted from ```_solution_old``` which is the solution from previous time step (?), say ```n```, that makes the evaluation as follows:

$$ \dot{u_{n-1}} = \frac{u_n - u_{n-2}}{\Delta t} $$

$$\ddot{u_{n-1}} =  \frac{u_n - 2u_{n-1} + u_{n-2}}{\Delta t^2} $$

This is different from the equation on website page https://mooseframework.inl.gov/source/timeintegrators/CentralDifference.html. I expect given the weak from with inertia term $\rho \ddot{u}$, the $\ddot{u}$ would first express as terms with $u_{n+1}, u_{n}, u_{n-1}$ and we solve for $u_{n+1}$, but it seems like here we get $\ddot{u}$ directly through previous values and fit it into the weak form? Thanks in advance.

**Comment by GiudGiud** [comment]
Hello

The documentation is wrong. 
As it it s implemented, the CentralDifference time integrator is an explicit time integrator, it does use u(t+dt) (u_n+1)

Guillaume

**Comment by hugary1995** [comment, marked as answer]
We followed these notes when implementing the central difference scheme: https://people.duke.edu/~hpgavin/StructuralDynamics/NumericalIntegration.pdf

@permcody might still have the "permission to include" from Prof. Gavin, so @GiudGiud we might just provide a link to these notes in our doco page.

**Reply by chunhuizhao478** [reply]
Hi Gray, thanks for sharing this. The equations in the note is consistent with the website. The part I still not sure about is the implementation in the code, it seems like ```u_old``` is not $u_{n+1}$, or it appears to be the case?
